packages/plugin-ext/src/main/browser/languages-main.ts

```diff
@@ -734,12 +736,20 @@ export class LanguagesMainImpl implements LanguagesMain, Disposable {
         }
         return {
             actions: actions.map(a => toMonacoAction(a)),
-            dispose: () => {
-                // TODO this.proxy.$releaseCodeActions(handle, cacheId);
-            }
+            dispose: () => this.proxy.$releaseCodeActions(handle, actions.map(a => a.cacheId))
         };
     }
 
+    protected async resolveCodeAction(handle: number, codeAction: monaco.languages.CodeAction, token: monaco.CancellationToken): Promise<monaco.languages.CodeAction> {
+        // The cacheId is kept in toMonacoAction when converting a received CodeAction DTO to a monaco code action
+        const cacheId = (codeAction as CodeAction).cacheId;
+        if (cacheId) {
```

Comment:
Minor: This could fail depending on an implementation detail, since `0` coerced into a boolean returns `false`, although `0` as a value of `cacheId` could point to a valid code action. If I understand this correctly, `cacheId` could be undefined in some circumstances, so we should check for `cacheId !== undefined` instead.